torchx/Makefile

```diff
@@ -22,7 +22,11 @@ check_libtorch:
 $(TORCHX_SO): c_src/torchx.cpp c_src/nx_nif_utils.hpp
 	@ mkdir -p $(PRIV_DIR)
 	@ mkdir -p $(CMAKE_BUILD_DIR)
-	@ ln -sf $(abspath $(LIBTORCH_DIR)/lib) $(PRIV_DIR)/libtorch
+	@ if [ "${BUILD_EMBEDDED}" = "false" ]; then \
+		ln -sf $(abspath $(LIBTORCH_DIR)/lib) $(PRIV_DIR)/libtorch ; \
+	else \
+		cp -a $(abspath $(LIBTORCH_DIR)/lib) $(PRIV_DIR)/libtorch ; \
+	fi
 	@ cd $(CMAKE_BUILD_DIR) && \
```

Comment:
Nice addition!

I think what remains to be solved is the optional toolchain cmake file, right?
Perhaps @fhunleth can help with this?

We could provide an empty toolchain file which is the default for the env var, and then it can be overriden as needed with the one provided by, say, Nerves

Reply:
I'm not entirely sure about how `elixir_make` works, but I think we can do it in `elixir_make` by adding a `toolchain`(or `cmake_toolchain`, or whatever suits best) option to `elixir_make`. And then, we can set the location of the optional toolchain CMake file in the top project (where torchx is a dependency), and fetch and use it here.

Reply:
This sounds nice, but I think that elixir_make is not aware of cmake at all, which is why we need to have a makefile on the top-level that calls cmake.

Reply:
Perhaps @josevalim can chip in on how we can use elixir_make with the cmake toolchain without actually having to make elixir_make itself aware of this.

It might be helpful to clarify how a given cmake toolchain file would interact with torchx, for instance.

Reply:
I changed my mind. I think maybe we should not make any changes to elixir_make, instead, we set an env var, say `CMAKE_TOOLCHAIN_FILE`, in the top project. And, as you said, we provide an empty toolchain file here by default and add `-D CMAKE_TOOLCHAIN_FILE=$(CMAKE_TOOLCHAIN_FILE)` to the Makefile.